src/Middleware/OutputCaching/src/OutputCachePolicyProvider.cs

```diff
@@ -0,0 +1,123 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.OutputCaching;
+
+internal sealed class OutputCachePolicyProvider
+{
+    private readonly OutputCacheOptions _options;
+
+    public OutputCachePolicyProvider(OutputCacheOptions options)
+    {
+        _options = options;
+    }
+
+    public bool HasPolicies(HttpContext httpContext)
+    {
+        if (_options.BasePolicies != null)
+        {
+            return true;
+        }
+
+        // Remove check
+        if (httpContext.Features.Get<IOutputCacheFeature>()?.Policies.Any() ?? false)
+        {
+            return true;
+        }
+
+        if (httpContext.GetEndpoint()?.Metadata.GetMetadata<IOutputCachePolicy>() != null)
```

Comment:
Another option here could be to use an internal metadata type. That way if OutputCaching want to add additional metadata to the endpoint that can't or shouldn't be described by the `IOutputCachePolicy`, we could still do so.

Reply:
As it was before an api review where it was asked to remove it since we didn't have any special properties on it. Though it allowed to also have the attribute wrapped for instance.